client/src/api/block_builder/input_selection/new/requirement/base_token.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use super::{InputSelection, OutputInfo, Requirement};
+use crate::{
+    block::output::{
+        unlock_condition::UnlockCondition, AliasOutputBuilder, FoundryOutputBuilder, NftOutputBuilder, Output, Rent,
+    },
+    error::{Error, Result},
+    secret::types::InputSigningData,
+};
+
+// TODO checked operations ?
+
+pub(crate) fn base_token_sums(selected_inputs: &[InputSigningData], outputs: &[OutputInfo]) -> (u64, u64) {
+    let inputs_sum = selected_inputs.iter().map(|input| input.output.amount()).sum::<u64>();
```

Comment:
The amount of SDRs is not subtracted here, if the required outputs with the SDR aren't created yet it might lead to some wrong result, because it could look as if more is "free" available even though it actually isn't

Looks fine how it's used now, but probably good to add a comment in case we want to use it also somewhere else

Reply:
I changed it. But instead of subtracting to inputs, I add to outputs. Achieves the same but I prefer it that way.